File: qqbot-persistence/src/main/java/com/mieai/qqbot/persistence/inbox/InboxPage.java
package com.mieai.qqbot.persistence.inbox;

import java.util.List;
import java.util.Objects;
import java.util.Optional;

/** One immutable page of Inbox events and an optional cursor for the next page. */
public record InboxPage(List<InboxEvent> events, Optional<String> nextCursor) {
    public InboxPage {
        events = List.copyOf(Objects.requireNonNull(events, "events must not be null"));
        nextCursor = Objects.requireNonNull(nextCursor, "nextCursor must not be null");
    }
}

File: qqbot-persistence/src/main/kotlin/com/mieai/qqbot/persistence/inbox/InboxPage.kt
package com.mieai.qqbot.persistence.inbox

/** One immutable page of Inbox events and an optional cursor for the next page. */
data class InboxPage private constructor(
    val events: List<InboxEvent>,
    val nextCursor: String?,
    private val normalized: Boolean,
) {
    constructor(events: List<InboxEvent>, nextCursor: String?) : this(
        events.toList(),
        nextCursor,
        true,
    )
}
